src/main/java/com/supermod/item/weapon/thunderbow/ThunderBowItem.java

```diff
@@ -0,0 +1,128 @@
+package com.supermod.item.weapon.thunderbow;
+
+import com.supermod.item.arrow.thunderarrow.ThunderArrowItem;
+import com.supermod.register.Registrable;
+import java.util.List;
+import javax.annotation.Nullable;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.stats.Stats;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.AbstractArrow;
+import net.minecraft.world.entity.projectile.Projectile;
+import net.minecraft.world.item.*;
+import net.minecraft.world.item.enchantment.EnchantmentHelper;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.event.ForgeEventFactory;
+import org.jetbrains.annotations.NotNull;
+
+
+public class ThunderBowItem extends BowItem implements Registrable {
+    public static final Properties PROPERTIES = new Item.Properties();
+
+    public ThunderBowItem() {
+        super(PROPERTIES);
+    }
+
+    public static String getNameStatic() {
+        return "thunder_bow";
+    }
+
+    @Override
+    public @NotNull UseAnim getUseAnimation(@NotNull ItemStack stack) {
+        return UseAnim.NONE;
+    }
+
+    private boolean is_enhanced_charge_attack(int charge) {
+        return charge >= 70;
+    }
+
+    @Override
+    public void releaseUsing(@NotNull ItemStack stack, @NotNull Level level, @NotNull LivingEntity entityLiving, int timeLeft) {
+        if (entityLiving instanceof Player player) {
+            ItemStack itemstack = player.getProjectile(stack);
+
+            if (!itemstack.isEmpty()) {
+                int i = this.getUseDuration(stack, entityLiving) - timeLeft;
+                i = ForgeEventFactory.onArrowLoose(stack, level, player, i, true);
+                if (i < 0) {
+                    return;
+                }
+
+                float f = getPowerForTime(i);
+                var isCharge = is_enhanced_charge_attack(i);
+                if (!((double) f < 0.1)) {
+                    List<ItemStack> list = draw(stack, itemstack, player);
+                    if (level instanceof ServerLevel serverlevel) {
+                        if (!list.isEmpty()) {
+                            float modifier = 3.0F * (isCharge ? 2 : 1);
+                            this.shoot(serverlevel, player, player.getUsedItemHand(), stack, list, f * modifier, f == 1.0F, isCharge, (LivingEntity) null);
```

Comment:
generally, too many params, try to shrink at least to 3, common param tuples extract into parameter object